Update to Protobuf 3.0.2

sourcedelica opened the pull request:
I updated to Protobuf 3.0.2 for gRPC which requires Protobuf 3.  I also made a few changes, mostly simplifications.  I'm not sure if they are things you'd want but I figured I'd PR it anyway so you could take a look.

I didn't test on Windows but I did test using Protobuf's CMake setup and it worked.  The biggest change was removing the customized CMake files.  I was concerned that they would would get out of sync with changes made by Google.   I don't know all of the reasons that these were split out, so I'm probably breaking something by removing them, but I'm interested to see if ultimately the build can do without them.

Also I tweaked `change_dylib_names.sh` so the package client wouldn't need to make any dylib changes beyond copying the dylibs to the executable directory.

Man, dylibs are a mess on Macos.

sourcedelica committed:
Update to Protobuf 3.0.2

sourcedelica committed:
Comment out the libstdc++11 check

sourcedelica committed:
Upgrade to Protobuf 3.3.0

memsharded commented:
Hi Eric!

Thanks for keeping working on this. I have merged it to my branch: release/3.3.0, and I am iterating over it, trying to pass CI. Specifically:
- Solved conflicts
- Windows: added Dprotobuf_MSVC_STATIC_RUNTIME to account for runtime
- Windows: Added suffix "d" to library names
- Linux: Using new AutoToolsbuildEnvironment instead of the deprecated ConfigureEnvironment
- Linux: use system_requirements() to install necessary system packages for building.

Lets see how CI goes. Please branch from this new branch if you intend to do further improvements. Thanks very much!

sourcedelica commented:
Cool!  I was doing some work on Mac and had to update some packages to
Apple Clang 8.1. That's great about your progress!  I will branch from
release/3.3.0


-- Eric

On Fri, May 5, 2017 at 5:43 AM, James <notifications@github.com> wrote:

> Hi Eric!
>
> Thanks for keeping working on this. I have merged it to my branch:
> release/3.3.0, and I am iterating over it, trying to pass CI. Specifically:
>
>    - Solved conflicts
>    - Windows: added Dprotobuf_MSVC_STATIC_RUNTIME to account for runtime
>    - Windows: Added suffix "d" to library names
>    - Linux: Using new AutoToolsbuildEnvironment instead of the deprecated
>    ConfigureEnvironment
>    - Linux: use system_requirements() to install necessary system
>    packages for building.
>
> Lets see how CI goes. Please branch from this new branch if you intend to
> do further improvements. Thanks very much!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/memsharded/conan-protobuf/pull/8#issuecomment-299424021>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEBjBJJZbvE6hwyFr3AYm5jIC3wBWanTks5r2u9NgaJpZM4LWccy>
> .
>
